runtime/interpreter/value.go

```diff
@@ -5649,6 +5910,47 @@ func (v *CompositeValue) TypeID() common.TypeID {
 	return v.Location.TypeID(v.QualifiedIdentifier)
 }
 
+func (v *CompositeValue) ConformsToDynamicType(dynamicType DynamicType) bool {
+	compositeType, ok := dynamicType.(CompositeDynamicType)
+	return ok && ValueConformsToSemaType(v, compositeType.StaticType)
```

Comment:
I wonder if we should include composites' fields' dynamic types in the composite dynamic type, just like we include the dynamic types of arrays' elements, and then check them here?

Reply:
Yup, that would be ideal.

If we have that, we can also get rid of, all the `value.conformsToSemaType()` methods, because this is the only entry point to that method.

Reply:
oh wait, that could be a problem when someone changes the value of a field. because we don't update the dynamic type upon the field updates. Wonder how it works for the arrays.